src/components/chat/Message.js

```diff
@@ -7,14 +7,21 @@ class Message extends Component {
     let { message, username } = this.props;
     let fromMe = message.from === username;
     let messageClass = fromMe ? 'chat-outgoing' : 'chat-incoming';
+
     let spaced = message.msg.replace(/:(\w+):/g, ':$1: ');
     let emojified = emoji.replace_colons(spaced);
-    let linked = md.renderInline(emojified);
+
+    // Convert `emoji.replace_colons`-generated <span> tags to Markdown
+    let emojiMD = emojified.replace(/<span class="emoji emoji-sizer" style="background-image:url\((\/static\/img\/emoji\/apple\/64\/.*?)\)" data-codepoints="(?:.*?)"><\/span>/g, '![emoji]($1)');
```

Comment:
@Moe-Shoman I know this looks crazy, but I couldn't get emoji image-containing HTML to safely render any other way; your `linkifycomplete` branch works fine on Mac because native emoji characters are being rendered there, not `<span>` tags.